### PR TITLE
plugins.ard_mediathek: fix / update

### PR DESCRIPTION
**Pull request by Billy2011**

### CODE REVIEW

File: src/streamlink/plugins/ard_mediathek.py [hunk 1]
```diff
@@ -100,24 +105,27 @@ def _get_streams(self):
...
                     parser_name = "SMIL"
-                elif stream_.endswith(".m3u8"):
+                elif ".m3u8" in stream_:
                     parser = self._get_hls_streams
                     parser_name = "HLS"
                 elif stream_.startswith("http"):
                     parser = self._get_http_streams
                     parser_name = "HTTP"
+                else:
+                    raise PluginError("Unexpected stream type: '{0}'".format(stream_))
```
Comment (back-to) [inline src/streamlink/plugins/ard_mediathek.py:122]: might result in an invalid raise

File: src/streamlink/plugins/ard_mediathek.py [hunk 2]
```diff
@@ -100,24 +105,27 @@ def _get_streams(self):
...
 
                 try:
                     for s in parser(stream):
                         yield s
                 except IOError as err:
-                    self.logger.error("Failed to extract {0} streams: {1}",
+                    raise PluginError("Failed to extract {0} streams: {1}",
```
Comment (back-to) [inline src/streamlink/plugins/ard_mediathek.py:128]: same here

File: src/streamlink/plugins/ard_mediathek.py [hunk 3]
```diff
@@ -15,8 +18,8 @@
     0: "144p"
 }
 
-_url_re = re.compile(r"http(s)?://(?:(\w+\.)?ardmediathek.de/tv|mediathek.daserste.de/)")
-_media_id_re = re.compile(r"/play/(?:media|config)/(\d+)")
+_url_re = re.compile(r"http(s)?://(?:(\w+\.)?ardmediathek.de/|mediathek.daserste.de/)")
```
Comment (back-to) [inline src/streamlink/plugins/ard_mediathek.py:21]: `mediathek.daserste.de` can be removed

File: src/streamlink/plugins/ard_mediathek.py (hunk 1 above)
Comment (Billy2011) [reply src/streamlink/plugins/ard_mediathek.py:122]: Why an invalid raise?

File: src/streamlink/plugins/ard_mediathek.py (hunk 2 above)
Comment (Billy2011) [reply src/streamlink/plugins/ard_mediathek.py:128]: If there is only a log message, you might not recognize the error.

File: src/streamlink/plugins/ard_mediathek.py (hunk 3 above)
Comment (Billy2011) [reply src/streamlink/plugins/ard_mediathek.py:21]: Is this solution not better?:
`_url_re = re.compile(r"http(s)?://(?:(\w+\.)?ardmediathek.de/((?!daserste).)*/|mediathek.daserste.de/)")`

Comment (back-to) [reply src/streamlink/plugins/ard_mediathek.py:21]: I don't think the domain is needed, it just redirects to the new domain.

Comment (Billy2011) [reply src/streamlink/plugins/ard_mediathek.py:21]: This Link can't be resolved with this plugin, this is better done with **ard_live** or if we rename it to **daserste** .
https://www.ardmediathek.de/daserste/sendung/babylon-berlin/Y3JpZDovL2Rhc2Vyc3RlLmRlL2JhYnlsb24tYmVybGlu/

Comment (Billy2011) [reply src/streamlink/plugins/ard_mediathek.py:21]: Sorry, I've test it with a wrong link.

Comment (Billy2011) [reply src/streamlink/plugins/ard_mediathek.py]: > mediathek.daserste.de can be removed

It fails with: [http://mediathek.daserste.de/live](http://mediathek.daserste.de/live)

**Pull request merged**